build_detail.go

```diff
@@ -0,0 +1,45 @@
+package voucher
+
+import (
+	"strings"
+)
+
+// BuildDetail is a type that describes the details/metadata info
+// related to a build
+type BuildDetail struct {
+	RepositoryURL string          `json:"repository"`
+	Commit        string          `json:"commit"`
+	BuildCreator  string          `json:"build_creator"`
+	BuildURL      string          `json:"build_url"`
+	ProjectID     string          `json:"project_id"`
+	Artifacts     []BuildArtifact `json:"artifacts"`
+}
+
+func (b *BuildDetail) String() string {
+	str := ""
+	if b.RepositoryURL != "" {
+		str += "RepositoryURL: " + b.RepositoryURL + "\n"
+	}
+	if b.Commit != "" {
+		str += "Commit: " + b.Commit + "\n"
+	}
+	if b.BuildCreator != "" {
+		str += "BuildCreator: " + b.BuildCreator + "\n"
+	}
+	if b.BuildURL != "" {
+		str += "BuildURL: " + b.BuildURL + "\n"
+	}
+	if b.ProjectID != "" {
+		str += "ProjectID: " + b.ProjectID + "\n"
+	}
+	str_artifacts := ""
```

Comment:
Alternatively, you could start this as a slice of strings!

```suggestion
	str_artifacts := make([]string, 0, len(b.Artifacts))
```

Currently we're expanding the `str_artifacts` every time we change it, whereas we could be allocating one big string at the end of the call when we use `strings.Join`.